// @seafile/seafile-calendar/lib/calendar/CalendarRightPanel.js
'use strict';

exports.__esModule = true;

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var CalendarRightPanel = function (_React$Component) {
  (0, _inherits3['default'])(CalendarRightPanel, _React$Component);

  function CalendarRightPanel(props) {
    (0, _classCallCheck3['default'])(this, CalendarRightPanel);

    var _this = (0, _possibleConstructorReturn3['default'])(this, _React$Component.call(this, props));

    _this.onSelect = function (value) {
      _this.setState({
        highlightTime: value
      });
      _this.props.onSelect(value);
    };

    _this.scrollUp = function () {
      _this.timeRef.current.scrollBy(0, -200);
    };

    _this.scrollDown = function () {
      _this.timeRef.current.scrollBy(0, 200);
    };

    _this.state = {
      highlightTime: _this.props.value || null
    };
    _this.timeRef = _react2['default'].createRef();
    return _this;
  }

  CalendarRightPanel.prototype.componentDidMount = function componentDidMount() {
    // The default time is 8, page scroll to 08:00
    this.timeRef.current.scrollTo(0, 34 * 16);
  };

  CalendarRightPanel.prototype.render = function render() {
    var _this2 = this;

    var _props = this.props,
        value = _props.value,
        prefixCls = _props.prefixCls,
        locale = _props.locale;

    var selectedDate = value.format().slice(0, 10);
    var times = [];
    for (var i = 0; i < 24; i++) {
      var str = (String(i) + ':00').padStart(5, '0');
      var str1 = (String(i) + ':30').padStart(5, '0');
      times.push(str);
      times.push(str1);
    }
    var highlight = this.state.highlightTime;
    var highlightTime = highlight ? highlight.format().slice(11, 16) : null;
    var isZhcn = locale && locale.today === '今天';
    return _react2['default'].createElement(
      'div',
      { className: prefixCls + '-right-panel' },
      _react2['default'].createElement(
        'div',
        { className: prefixCls + '-right-panel-header', onClick: this.scrollUp },
        _react2['default'].createElement('span', null)
      ),
      _react2['default'].createElement(
        'div',
        { className: prefixCls + '-right-panel-body', ref: this.timeRef },
        _react2['default'].createElement(
          'ul',
          null,
          times.map(function (time) {
            var current = (0, _moment2['default'])(selectedDate + ' ' + time);
            current = isZhcn ? current.locale('zh-cn') : current.locale('en-gb');
            return _react2['default'].createElement(
              'li',
              {
                key: time,
                onClick: _this2.onSelect.bind(_this2, current),
                className: '' + (highlightTime === time ? prefixCls + '-selected-time' : '')
              },
              time
            );
          })
        )
      ),
      _react2['default'].createElement(
        'div',
        { className: prefixCls + '-right-panel-footer', onClick: this.scrollDown },
        _react2['default'].createElement('span', null)
      )
    );
  };

  return CalendarRightPanel;
}(_react2['default'].Component);

CalendarRightPanel.propTypes = {
  prefixCls: _propTypes2['default'].string,
  value: _propTypes2['default'].object,
  onSelect: _propTypes2['default'].func,
  locale: _propTypes2['default'].object
};
exports['default'] = CalendarRightPanel;
module.exports = exports['default'];
